### ISSUE TITLE
Improve interface of lens.find method

### ISSUE THREAD

**Issue by karelklima** [karelklima/ldkit#52]
The interface should enable user to do more powerful queries, order results, and find specific resources more easily.

Prisma could be a good inspiration in what to accomplish:
```typescript
const results = await prisma.post.findMany({
  skip: 200,
  take: 20,
  where: {
    email: {
      contains: 'Prisma',
    },
  },
  orderBy: {
    title: 'desc',
  },
})
```

**Comment by karelklima** [comment]
As for the operators = they should correspond to sparql boolean FILTER functions, e.g. `regex`, `contains`, `lang`.

Alternatively, there could be a `filter` definition that would translate directly to SPARQL, e.g.:
```typescript
{
  where: {
    someNumber: {
      filter: "? > 10",
    },
  },
}
```

**Comment by karelklima** [comment]
Partially resolved by #65.

**Comment by karelklima** [comment]
Ordering is out of scope and not really aligned with the Linked Data concepts supported by LDkit. Closing.
